git_machete/github.py

```diff
@@ -321,7 +321,7 @@ def get_parsed_github_remote_url(url: str, remote: str) -> Optional[RemoteAndOrg
         if match:
             return RemoteAndOrganizationAndRepository(remote=remote,
                                                       organization=match.group(1),
-                                                      repository=match.group(2))
+                                                      repository=match.group(2) if match.group(2)[-4:] != '.git' else match.group(2)[:-4])
```

Comment:
Pls add a comment in code about impossibility of `.git`-suffixed repo names so that this knowledge isn't lost

Plus extract this param to a local variable